### PR TITLE
Update to tauri 2

### PR DESCRIPTION
**Pull request by ImUrX**
This requires some beta testing, and supposedly tauri 2 is in alpha. But I kinda need it for flatpak so I'm gonna maintain this PR

### CODE REVIEW

**Comment by ImUrX** [comment]
Logs from server are received as u8 arrays, need to fix that

File: gui/src-tauri/src/main.rs [hunk 1]
```diff
@@ -92,10 +92,11 @@ fn main() {
 
 	// Spawn server process
 	let exit_flag = Arc::new(AtomicBool::new(false));
-	let mut backend: Option<CommandChild> = None;
+	let backend = Arc::new(Mutex::new(Option::<CommandChild>::None));
```
Comment (TheButlah) [inline gui/src-tauri/src/main.rs:95]: I'm not very happy with this, but its out of the scope of this PR. I think I will submit a PR to refactor this, its becoming messier and messier and this code should all be dead simple.

**Review by TheButlah — APPROVED** [review]

**Comment by TheButlah** [comment]
Approved, but I did not run it.

File: gui/src-tauri/src/main.rs (hunk 1 above)
Comment (ImUrX) [reply gui/src-tauri/src/main.rs:95]: thats honestly understandable, its a hack rn

File: gui/src/i18n/config.tsx [hunk 2]
```diff
@@ -109,9 +109,7 @@ export const langs = [
 // AppConfig path: https://docs.rs/tauri/1.2.4/tauri/api/path/fn.config_dir.html
 // We doing this only once, don't want an override check to be done on runtime,
 // only on launch :P
-const overrideLangExists = exists(OVERRIDE_FILENAME, {
-  dir: BaseDirectory.AppConfig,
-}).catch(() => false);
```
Comment (ButterscotchV) [inline gui/src/i18n/config.tsx:114]: This looks like it changes behaviour, what's this change done for?

Comment (ImUrX) [reply gui/src/i18n/config.tsx:114]: it doesnt, just that exists doesn't have a dir parameter

**Comment by loucass003** [comment]
Sync to main so i test and approve

**Review by loucass003 — APPROVED** [review]
LGTM

**Review by Eirenliel — APPROVED** [review]
Merge after 0.9.0 release

**Review by Louka3000 — APPROVED** [review]

**Pull request merged**